Source/quests.cpp

```diff
@@ -885,4 +873,14 @@ bool QuestStruct::IsAvailable()
 	return true;
 }
 
+bool QuestStruct::EntranceBoundaryContains(Point position) const

```

Comment:
I think this is the same shape as used for town portal portals, potentially useful to pull this out of the QuestStruct class.

Reply:
Indeed CheckTown and CheckRportal have the same checks

Reply:
Plz see https://github.com/diasurgical/devilutionX/pull/2845